src/analysis/normed_space/box_subadditive.lean

```diff
@@ -0,0 +1,541 @@
+    t.piecewise_le_of_le_of_le le_rfl I.midpoint_mem.2,
+  refine ⟨⟨_, _, _, set.subset.trans (set.Icc_subset_Icc hl hr) I.sub, ht⟩, hl, hr, λ i, _⟩,
+  { exact t.piecewise_le_piecewise I.midpoint_mem.2 I.midpoint_mem.1 },
+  { by_cases hi : i ∈ t; simp [hi, div_eq_inv_mul] }
+end
+
+/-- An auxiliary definition for `box_subadditive_on.eq_zero_of_forall_eventually_le_mul`:
+a decreasing sequence of subboxes `[l n, u n]` such that `c * g (l n) (u n) < f (l u) (u n)`. -/
+def seq (hf : box_subadditive_on f s) (hg : box_supadditive_on g s) (I₀ : subbox_mul_lt s f g c)
+  (n : ℕ) : subbox_mul_lt s f g c :=
+(λ I, hf.next hg I)^[n] I₀
+
+/-- An auxiliary definition for `box_subadditive_on.eq_zero_of_forall_eventually_le_mul`:
+the limit point of the sequence `box_subadditive_on.seq hf hg I hI`. -/
+def fix (hf : box_subadditive_on f s) (hg : box_supadditive_on g s) (I : subbox_mul_lt s f g c) :
+  ι → ℝ :=
+⨆ n, (seq hf hg I n).left
+
+@[simp] lemma seq_zero (hf : box_subadditive_on f s) (hg : box_supadditive_on g s)
+  (I : subbox_mul_lt s f g c) :
+  seq hf hg I 0 = I := rfl
+
+@[simp] lemma seq_succ (hf : box_subadditive_on f s) (hg : box_supadditive_on g s)
+  (I : subbox_mul_lt s f g c) (n : ℕ) :
+  seq hf hg I (n + 1) = next hf hg (seq hf hg I n) :=
+iterate_succ_apply' _ _ _
+
+@[simp] lemma seq_right_sub_left_apply (hf : box_subadditive_on f s) (hg : box_supadditive_on g s)
+  (I : subbox_mul_lt s f g c) (n : ℕ) (i : ι) :
+  (seq hf hg I n).right i - (seq hf hg I n).left i = (I.right i - I.left i) / 2 ^ n :=
+begin
+  induction n with n ihn, { simp },
+  rw [seq_succ, (next hf hg _).coe_prop.2.2, ihn, pow_succ', div_div_eq_div_mul]
+end
+
+@[simp] lemma seq_right_sub_left (hf : box_subadditive_on f s) (hg : box_supadditive_on g s)
+  (I : subbox_mul_lt s f g c) (n : ℕ) :
+  (seq hf hg I n).right - (seq hf hg I n).left = ((1 / 2) ^ n : ℝ) • (I.right - I.left) :=
+funext $ λ i, (seq_right_sub_left_apply hf hg I n i).trans $
+  by simp only [div_eq_inv_mul, pi.smul_apply, pi.sub_apply, smul_eq_mul, mul_one, inv_pow']
+
+lemma dist_seq_left_right (hf : box_subadditive_on f s) (hg : box_supadditive_on g s)
+  (I : subbox_mul_lt s f g c) (n : ℕ) :
+  dist (seq hf hg I n).left (seq hf hg I n).right = dist I.left I.right / 2 ^ n :=
+by rw [dist_comm, dist_eq_norm, seq_right_sub_left, norm_smul, ← dist_eq_norm, dist_comm, mul_comm,
+  one_div, inv_pow', normed_field.norm_inv, normed_field.norm_pow, real.norm_eq_abs,
+  abs_of_pos (@zero_lt_two ℝ _ _), div_eq_mul_inv]
+
+lemma seq_left_mono (hf : box_subadditive_on f s) (hg : box_supadditive_on g s)
+  (I : subbox_mul_lt s f g c) :
+  monotone (λ n, (seq hf hg I n).left) :=
+monotone_of_monotone_nat $ λ n, by simp only [seq_succ, (next _ _ _).coe_prop.1]
+
+lemma seq_right_mono (hf : box_subadditive_on f s) (hg : box_supadditive_on g s)
+  (I : subbox_mul_lt s f g c) :
+  ∀ ⦃m n⦄, m ≤ n → (seq hf hg I n).right ≤ (seq hf hg I m).right :=
```

Comment:
We will really need to add the definition `antimono` to mathlib at some point (but not in this PR)